.gitlab/llnl-ci-spack.yml

```diff
@@ -68,8 +64,9 @@ stages:
     - for i in {1..3}; do
         spack env activate ${SPACK_ENV_NAME} && break || sleep 5;
       done
-    - spack uninstall -y exaworks%${COMPILER} py-pytest%${COMPILER} || true
-    - spack add exaworks%${COMPILER} ^python@3.9 py-pytest%${COMPILER}
+    - spack uninstall -y exaworks py-pytest || true
+    - spack add exaworks ^python@3.9 py-pytest
+    - spack config add concretizer:unify:when_possible
```

Comment:
should we remove this? or you do think this configuration is helpful? -> if to keep then it should be in the `.spack_env_setup` job

Reply:
I was getting the following error on Quartz and Ruby:
```
ncurses@=6.4%gcc@=12.1.1~symlinks+termlib abi=none build_system=autotools arch=linux-rhel8-broadwell
flux-core requires lua version 5.1:5.3, but spec asked for =5.4.4. You could consider setting `concretizer:unify` to `when_possible` or `false` to allow multiple versions of some packages.
```

After applying `spack config add concretizer:unify:when_possible`, it fixed the error. I think it's a valid change that might mitigate any future dependency resolution issues.

I can move it over to the spack_env_setup job

Reply:
ok, then it will update the current setting
https://github.com/ExaWorks/SDK/blob/629b95e5eb358c8a69266fd8101a642e04017d9c/.gitlab/llnl-ci-spack.yml#L52